plugins/pluginmanagers/lazy.nix

```diff
@@ -17,105 +17,113 @@ nixvim.neovim-plugin.mkNeovimPlugin {
+                Lazy uses several heuristics to determine the plugin's MAIN module automatically based on the plugin's name.
+                See also opts. To use the default implementation without opts set config to true.
+              '';
+
+              main = helpers.mkNullOrOption str ''
+                You can specify the main module to use for config() and opts(),
+                in case it can not be determined automatically. See config()
+              '';
+
+              submodules = helpers.defaultNullOpts.mkBool true ''
+                When false, git submodules will not be fetched.
+                Defaults to true
+              '';
+
+              event =
+                with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+
+              cmd =
+                with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on command";
+
+              ft =
+                with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on filetype";
+
+              keys =
+                with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on key mapping";
+
+              module = helpers.mkNullOrOption (enum [ false ]) ''
+                Do not automatically load this Lua module when it's required somewhere
+              '';
+
+              priority = helpers.mkNullOrOption number ''
+                Only useful for start plugins (lazy=false) to force loading certain plugins first.
+                Default priority is 50. It's recommended to set this to a high number for colorschemes.
               '';
-          };
-        });
+
+              optional = helpers.defaultNullOpts.mkBool false ''
+                When a spec is tagged optional, it will only be included in the final spec,
+                when the same plugin has been specified at least once somewhere else without optional.
+                This is mainly useful for Neovim distros, to allow setting options on plugins that may/may not be part
+                of the user's plugins
+              '';
+
+              opts =
+                with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (attrsOf anything)) ''
+                  opts should be a table (will be merged with parent specs),
+                  return a table (replaces parent specs) or should change a table.
+                  The table will be passed to the Plugin.config() function.
+                  Setting this value will imply Plugin.config()
+                '';
+            };
+
+            config.name = lib.mkIf (config.pkg != null) (lib.mkDefault "${lib.getName config.pkg}");
```

Comment:
This is because of the type coercion.

`types.coercedTo` normalizes the final/merged option value to the "final" type, in this case that means any `types.package` definitions will be converted to `{ pkg = definition; }` in the final config value.

> Personally I prefer the non type coercion approach

Gotcha. Is that because of [`9d9dc58` (#2082)](https://github.com/nix-community/nixvim/pull/2082/commits/9d9dc581c00c6670686a41866d3fa3adaa57ca85)? Or because you don't like the idea of merging definitions in this way?